cmd/kubeadm/app/preflight/checks.go

```diff
@@ -402,9 +402,13 @@ func (HostnameCheck) Name() string {
 }
 
 // Check validates if hostname match dns sub domain regex.
+// Also check hostname length
 func (hc HostnameCheck) Check() (warnings, errorList []error) {
-	klog.V(1).Infoln("checking whether the given node name is reachable using net.LookupHost")
+	klog.V(1).Infoln("checking whether the given node name is valid and reachable using net.LookupHost")
 
+	if len(hc.nodeName) > 63 {
+		warnings = append(warnings, errors.Errorf("hostname \"%s\" should be no more than 63 characters. Change hostname or use --node-name flag to override it", hc.nodeName))
+	}
```

Comment:
i would like to understand more about the kubelet requirements before applying such a change.
https://github.com/kubernetes/kubernetes/issues/99156

aren't we supposed to use IsQualifiedName() instead of pining to 63?

Reply:
> #99156

this is closed, until further notice.
does the kubelet actually impose validation and limitation to the hostnames the user can pass to it?

Reply:
kubelet doesn't validate it, but hostname is used in some other features like node-label selector and static-pod suffix.

If it is too long, something may not work. So I think we should add a warning here. 
As this is not strictly limited, it is not an error.

Reply:
true, we are adding a warning here in kubeadm and that's fine, but i'm wondering about a few things:
- does the bootstrap finish OK (i.e. node joins the cluster)?
- can pods still schedule on this node?
i guess i can try some of these things myself..

i'm asking because maybe the kubelet should also error out on long names (separate PR).
but if some of that partially works such a change in the kubelet would need an "action required" because otherwise it will break some users.

Reply:
Add to my todo list on kubelet/sig-node